<?xml version='1.0'?>

<launch>
  <!-- Input Arguments -->
  
  <!-- For closed-loop controller - Aggressive setup -->
  <!-- <arg name="velocity_max" default="12.0" />
  <arg name="acc_long_max" default="1.5" />
  <arg name="acc_lat_max" default="8.0" />
  <arg name="dec_long_max" default="2.0" /> -->
  
  <!-- For closed-loop controller - Normal setup -->
  <arg name="velocity_max" default="6.0" />
  <arg name="acc_long_max" default="1.0" />
  <arg name="acc_lat_max" default="4.0" />
  <arg name="dec_long_max" default="2.0" />

  <!-- For open-loop controller -->
  <!-- <arg name="velocity_max" default="8.0" />
  <arg name="acc_long_max" default="1.0" /> 
  <arg name="dec_long_max" default="1.0" />
  <arg name="acc_lat_max" default="2.0" /> -->
  
  <arg name="path_type" default="straight" /> <!-- Type of path: 'straight', 'circular', 'lane_change', 'infinity' -->
  <arg name="publish_heading" default="true" /> <!-- Set to true if using closed-loop controlelr -->

  <!-- Launch Path Planner Node -->
  <node pkg="path_planning_pkg" type="offline_planner.py" name="offline_planner" output="screen">
    <!-- Path related parameters -->
    <param name="path_type" type="str" value="$(arg path_type)"/>
    <param name="path_length" value="6.283185307"/>
    <param name="path_length_straight" value="15.0"/>
    <param name="path_length_curve_x" value="15.0"/>
    <param name="path_radius" value="6.0"/> <!-- Circular or infinity -->
    <!-- <param name="step_size" value="0.034906585"/> -->
    <param name="step_size" value="0.2"/>
    <param name="rotate_ccw" value="false"/>
    <param name="lane_width" value="3.7"/>
    <param name="turn_left" value="true"/>
    <param name="position_current" type="str" value="[1.1345, 0.0]"/>
    <!-- <param name="position_current" type="str" value="[-73.8655, 0.0]"/> -->
    <param name="publish_heading" value="$(arg publish_heading)" />
    
    <!-- Velocity profile related parameters -->
    <param name="velocity_max" value="$(arg velocity_max)" />
    <param name="acc_long_max" value="$(arg acc_long_max)" />
    <param name="acc_lat_max" value="$(arg acc_lat_max)" />
    <param name="dec_long_max" value="$(arg dec_long_max)" />
  </node>

</launch>

